### PR TITLE
remove stackdriver, elastic task pages

### PR DESCRIPTION
**Pull request by kbhawkey**
Investigating whether these task pages are considered third party content.
General cleanup and style updates.
Related to issue #20232.

### CODE REVIEW

File: content/en/docs/concepts/cluster-administration/logging.md [hunk 1]
```diff
@@ -133,7 +133,10 @@ Because the logging agent must run on every node, it's common to implement it as
 
 Using a node-level logging agent is the most common and encouraged approach for a Kubernetes cluster, because it creates only one agent per node, and it doesn't require any changes to the applications running on the node. However, node-level logging _only works for applications' standard output and standard error_.
 
-Kubernetes doesn't specify a logging agent, but two optional logging agents are packaged with the Kubernetes release: [Stackdriver Logging](/docs/tasks/debug-application-cluster/logging-stackdriver/) for use with Google Cloud Platform, and [Elasticsearch](/docs/tasks/debug-application-cluster/logging-elasticsearch-kibana/). You can find more information and instructions in the dedicated documents. Both use [fluentd](https://www.fluentd.org/) with custom configuration as an agent on the node.
+Kubernetes doesn't specify a logging agent, but two optional logging agents are integrated with
+Kubernetes: [Stackdriver Logging](https://cloud.google.com/stackdriver/docs) and
+[Elasticsearch](https://www.elastic.co/elastic-stack). Both logging agents use
+[fluentd](https://www.fluentd.org/) with custom configuration as an agent on the node.
```
Comment (sftim) [inline content/en/docs/concepts/cluster-administration/logging.md:139]: I think this is dated. By now, several more products work with Kubernetes for logging. It's a long list.

File: content/en/docs/tutorials/stateful-application/zookeeper.md [hunk 2]
```diff
@@ -559,8 +559,8 @@ You can view application logs written to standard out or standard error using `k
 ```
 
 Kubernetes supports more powerful, but more complex, logging integrations
-with [Stackdriver](/docs/tasks/debug-application-cluster/logging-stackdriver/)
-and [Elasticsearch and Kibana](/docs/tasks/debug-application-cluster/logging-elasticsearch-kibana/).
+with [Stackdriver](https://cloud.google.com/stackdriver/docs)
+and [Elasticsearch and Kibana](https://www.elastic.co/elastic-stack).
```
Comment (sftim) [inline content/en/docs/tutorials/stateful-application/zookeeper.md:563]: I think this is dated. By now, several more products work with Kubernetes for logging. It's a long list.

We should tell people that Kubernetes leaves it up to the cluster operator to select a logging solution and integrate it with their cluster or application.

File: content/en/docs/tutorials/stateful-application/zookeeper.md [hunk 3]
```diff
@@ -559,8 +559,8 @@ You can view application logs written to standard out or standard error using `k
...
 For cluster level log shipping and aggregation, consider deploying a [sidecar](https://kubernetes.io/blog/2015/06/the-distributed-system-toolkit-patterns)
 container to rotate and ship your logs.
```
Comment (sftim) [inline content/en/docs/tutorials/stateful-application/zookeeper.md:565]:
```suggestion
For cluster level log shipping and aggregation, consider deploying a [sidecar](/docs/concepts/cluster-administration/logging/#using-a-sidecar-container-with-the-logging-agent)
```

File: content/en/docs/concepts/cluster-administration/logging.md [hunk 4]
```diff
@@ -133,7 +133,10 @@ Because the logging agent must run on every node, it's common to implement it as
 
 Using a node-level logging agent is the most common and encouraged approach for a Kubernetes cluster, because it creates only one agent per node, and it doesn't require any changes to the applications running on the node. However, node-level logging _only works for applications' standard output and standard error_.
```
Comment (sftim) [inline content/en/docs/concepts/cluster-administration/logging.md:134]: Is this still true?

To me, the two options are:
- containers write stdout and stderr, but with no agreed format. A node-level agent collects these logs and forwards them for aggregation
- applications in containers send logs, possibly via a sidecar, to an aggregation service. The aggregator could be running inside the cluster, or external to it.
  -  node-level logging can happen, separately, to report things such as system upgrades, hardware failure events, etc


both options have their fans. I'm not convinced the argument is settled either way, nor that it needs to be.

**Review by sftim — COMMENTED** [review]
My thoughts on this

File: content/en/docs/concepts/cluster-administration/logging.md (hunk 1 above)
Comment (kbhawkey) [reply content/en/docs/concepts/cluster-administration/logging.md:139]: Yes, I agree. I could continue to clean up in this page.

File: content/en/docs/concepts/cluster-administration/logging.md (hunk 4 above)
Comment (kbhawkey) [reply content/en/docs/concepts/cluster-administration/logging.md:134]: I have not looked at this page in detail. I originally wanted feedback on deleting the task pages.
If that is the right thing to do, then go ahead and clean up the associated references and pages.
Let me see if I can update a bit more.

File: content/en/docs/tutorials/stateful-application/zookeeper.md (hunk 2 above)
Comment (kbhawkey) [reply content/en/docs/tutorials/stateful-application/zookeeper.md:563]: Sure. I can update this.

**Comment by celestehorgan** [comment]
General question: why keep the page on Apache Zookeeper?

**Comment by sftim** [comment]
> Why keep the page on Apache Zookeeper?

It's useful to have an example of _some_ distributed application with roughly the properties of ZooKeeper. We'd want to have a tutorial where readers can learn about  [surviving maintenance](https://kubernetes.io/docs/tutorials/stateful-application/zookeeper/#surviving-maintenance) for example.

**Comment by celestehorgan** [comment]
Fair. I suppose this falls under the same situation as the networking example (if I remember correctly) used for the CKA..

**Comment by kbhawkey** [comment]
> 
> 
> General question: why keep the page on Apache Zookeeper?

I think the Zookeeper tutorial could be updated/cleaned up a bit, but it is an example of a distributed stateful application.
In addition, Apache Zookeeper is open source.

**Comment by sftim** [comment]
> I suppose this falls under the same situation as the networking example (if I remember correctly) used for the CKA.

It's similar but _less_ strong. A cluster with no networking is not much of a cluster, so network plugins are essential. Teaching readers how Kubernetes recommends deploying distributed apps that have their own clustering mechanisms, that's useful but people who don't know that can still make use of their cluster for other tasks - such as bin-packing stateless workloads.

If something else like the distributed app (Zookeeper) page comes up, it might need consideration on a case-by-case basis but still guided by the content guide & relevant KEP.

**Comment by kbhawkey** [comment]
👋 I read through the Zookeeper tutorial page again. I have previously (and briefly) used Zookeeper.
I wonder if the docs could provide a less complicated tutorial? I find the setup restrictive.
The tutorial also includes and relies upon a fair amount of Zookeeper specific information to walk the reader
through the content.
Any thoughts on another technology/tool/topic that could serve as a replacement tutorial? I wonder how often this page is read.

**Comment by sftim** [comment]
> I wonder if the docs could provide a less complicated tutorial?

The tutorial needs to be a bit complicated, ideally with justification. If you had a simple app it might not need a replicated statefulset.

> Any thoughts on another technology/tool/topic that could serve as a replacement tutorial? I wonder how often this page is read.

How about Apache Kafka?

**Comment by sftim** [comment]
BTW Kafka [no longer depends on Zookeeper](https://www.infoq.com/presentations/kafka-zookeeper/)

**Comment by sftim** [comment]
Relevant to issue #25111

**Review by sftim — COMMENTED** [review]
/lgtm

**Comment by kbhawkey** [comment]
👋 @x13n Would you have time to review these changes?

File: content/en/docs/concepts/cluster-administration/logging.md [hunk 5]
```diff
@@ -44,73 +45,73 @@ To fetch the logs, use the `kubectl logs` command, as follows:
...
+You can use `kubectl logs --previous` to retrieve logs from a previous instantiation of a container. If your pod has multiple containers, specify which container's logs you want to access by appending a container name to the command. See the [`kubectl logs` documentation](/docs/reference/generated/kubectl/kubectl-commands#logs) for more details.
 
 ## Logging at the node level
 
 ![Node level logging](/images/docs/user-guide/logging/logging-node-level.png)
 
-Everything a containerized application writes to `stdout` and `stderr` is handled and redirected somewhere by a container engine. For example, the Docker container engine redirects those two streams to [a logging driver](https://docs.docker.com/engine/admin/logging/overview), which is configured in Kubernetes to write to a file in json format.
+A container engine handles and redirects any output generated to a containerized application's `stdout` and `stderr` streams.
+For example, the Docker container engine redirects those two streams to [a logging driver](https://docs.docker.com/engine/admin/logging/overview), which is configured in Kubernetes to write to a file in JSON format.
```
Comment (sftim) [inline content/en/docs/concepts/cluster-administration/logging.md:65]: Aside: the written format is actually [JSONlines](https://jsonlines.org/) (the Docker documentation is also incorrect).
Each log entry is a JSON document, separated by a newline.

File: content/en/docs/concepts/cluster-administration/logging.md [hunk 6]
```diff
@@ -44,73 +45,73 @@ To fetch the logs, use the `kubectl logs` command, as follows:
...
+`kubectl logs` returns the latest log file which in this example is an empty response.
 {{< /note >}}
 
-[cosConfigureHelper]: https://github.com/kubernetes/kubernetes/blob/{{< param "githubbranch" >}}/cluster/gce/gci/configure-helper.sh
 ### System component logs
 
 There are two types of system components: those that run in a container and those
 that do not run in a container. For example:
 
 * The Kubernetes scheduler and kube-proxy run in a container.
```
Comment (sftim) [inline content/en/docs/concepts/cluster-administration/logging.md:103]: Aside: this is a common configuration, but not required.

**Review by sftim — COMMENTED** [review]
/lgtm

File: content/en/docs/concepts/cluster-administration/logging.md (hunk 6 above)
Comment (kbhawkey) [reply content/en/docs/concepts/cluster-administration/logging.md:103]: OK. I'll leave this to another PR.

File: content/en/docs/concepts/cluster-administration/logging.md (hunk 5 above)
Comment (kbhawkey) [reply content/en/docs/concepts/cluster-administration/logging.md:65]: OK. Is there more information about the use of JSON Lines in Kubernetes? I am happy to update if I can find more information.
What are your thoughts about removing the two task pages?
Is there a need to create new task pages?
I am looking at these links (no mention of JSON Lines):
https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/1602-structured-logging#json-output-format
https://kubernetes.io/docs/concepts/cluster-administration/system-logs/
This pull request is mostly focused on cleanup.

Comment (celestehorgan) [reply content/en/docs/concepts/cluster-administration/logging.md:65]: gotchu with an approve, tim!

**Pull request merged**